Makefile

```diff
@@ -604,6 +607,10 @@ $(OBJDIR)/%.long-double.pic.o: %.c include_dirs
 	@mkdir -p "$(@D)"
 	$(CC) $(CFLAGS) -MD -MP -o $@ -c $<
 
+$(OBJDIR)/preview2_component_type.pic.o $(OBJDIR)/preview2_component_type.o: $(LIBC_BOTTOM_HALF_SOURCES)/preview2_component_type.o
+	@mkdir -p "$(@D)"
+	cp $< $@
```

Comment:
Is copying needed, or would it be enough to just add `$(LIBC_BOTTOM_HALF_SOURCES)/preview2_component_type.o` to `LIBC_OBJS`?

Reply:
Sorry, I saw this and then forgot to follow up.  I'll see if I can eliminate the copy.  I think I was concerned adding a already-generated object file to `LIBC_OBJS` would break a pattern rule somewhere, but we'll see...

Reply:
I took another look at this, and I think eliminating the copy would be more trouble than it's worth since the pattern rules for building the .o files for libc.a and libc.so each need their own version (i.e. the .o and the .pic.o, respectively).  The file is only 11KB so I don't think the copy adds noticeable bloat to the build.